viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java

```diff
@@ -0,0 +1,210 @@
+
+    mInputGraphicsOverlay = new GraphicsOverlay();
+    mResultGraphicsOverlay = new GraphicsOverlay();
+
+    // inflate MapView from layout
+    mMapView = (MapView) findViewById(R.id.mapView);
+    // create a map with the BasemapType topographic
+    ArcGISMap map = new ArcGISMap(Basemap.Type.TOPOGRAPHIC, 45.3790902612337, 6.84905317262762, 12);
+    // set the map to be displayed in this view
+    mMapView.setMap(map);
+
+    // renderer for graphics overlays
+    SimpleMarkerSymbol pointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.RED, 10);
+    SimpleRenderer renderer = new SimpleRenderer(pointSymbol);
+    mInputGraphicsOverlay.setRenderer(renderer);
+
+    int fillColor = Color.argb(120, 226, 119, 40);
+    FillSymbol fillSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, fillColor, null);
+    mResultGraphicsOverlay.setRenderer(new SimpleRenderer(fillSymbol));
+
+    // add graphics overlays to the map view
+    mMapView.getGraphicsOverlays().add(mResultGraphicsOverlay);
+    mMapView.getGraphicsOverlays().add(mInputGraphicsOverlay);
+
+    mGeoprocessingTask = new GeoprocessingTask(getString(R.string.viewshed_service));
+
+    mMapView.setOnTouchListener(new DefaultMapViewOnTouchListener(getApplicationContext(), mMapView) {
+      @Override public boolean onSingleTapConfirmed(MotionEvent e) {
+        android.graphics.Point screenPoint = new android.graphics.Point(Math.round(e.getX()), Math.round(e.getY()));
+        Point mapPoint = mMapView.screenToLocation(screenPoint);
+        addGraphicForPoint(mapPoint);
+        calculateViewshedAt(mapPoint);
+        return super.onSingleTapConfirmed(e);
+      }
+    });
+  }
+
+  /**
+   * Adds a graphic at the chosen mapPoint.
+   *
+   * @param point in MapView coordinates.
+   */
+  private void addGraphicForPoint(Point point) {
+    // remove existing graphics
+    mInputGraphicsOverlay.getGraphics().clear();
+
+    // new graphic
+    Graphic graphic = new Graphic(point);
+
+    // add new graphic to the graphics overlay
+    mInputGraphicsOverlay.getGraphics().add(graphic);
+  }
+
+  /**
+   * Uses the given point to create a FeatureCollectionTable which is passed to performGeoprocessing.
+   *
+   * @param point in MapView coordinates.
+   */
+  private void calculateViewshedAt(Point point) {
+    // remove previous graphics
+    mResultGraphicsOverlay.getGraphics().clear();
+
+    // cancel any previous job
+    if (mGeoprocessingJob != null) {
+      mGeoprocessingJob.cancel();
+    }
+
+    // TODO - This feels like a bit of a hack. Using new ArrayList<Field>() as the first parameter of new
+    // TODO - FeatureCollectionTable(new ArrayList<Field>(), ...) throws an error
+    Field field = Field.createInteger("placeholder", null);
+
+    // create feature collection table for point geometry
+    final FeatureCollectionTable featureCollectionTable = new FeatureCollectionTable(Collections.singletonList(field),
```

Comment:
Developer pattern should be to create a the table schema by defining the a set of fields:

```java
List<Field> fields = new ArrayList<>(1);
// create field with same alias as name
Field field = Field.createString("observer", "", 8);
fields.add(field);
final FeatureCollectionTable featureCollectionTable = new FeatureCollectionTable(fields,
    GeometryType.POINT, point.getSpatialReference());
```